File: Python/src/dss/collections/queue.py
class Queue:
  def __init__(self) -> None:
    self.__queue = []

  def __len__(self):
    return len(self.__queue)

  def enqueue(self, value):
    self.__queue.append(value)
    return True

  def dequeue(self):
    return self.__queue.pop(0)

File: NodeJS/src/collections/queue.js
export { Queue };

/**
 * @description A array based queue implementation
 */

class Queue {
  #QUEUE;
  constructor() {
    this.#QUEUE = [];
  }

  /** Access */
  peek() {
    return this.#QUEUE[0];
  }

  /** Insertions */
  enqueue(v) {
    this.#QUEUE.push(v);
    return true;
  }

  /** Deletions */
  deQueue() {
    return this.#QUEUE.shift();
  }
}
